### ISSUE TITLE
TextClip generating OSError when string is empty ""

### ISSUE THREAD

**Issue by gooday2die** [Zulko/moviepy#1800]
Hello, before I start, thank you for this simple yet powerful library. While I was making an video using moviepy, there was a case that my script generated an empty text(`""`) for `TextClip` as its argument `text`. The script and moviepy runs smoothly with all other cases, however when `TextClip` gets its `text` argument as `""`, it crashes. So I am reporting this issue here. 

#### Expected Behavior
I am not sure what to expect with empty string `""`. 

However in my guess, there might be two options with the expected behaviors:
1. Generating a `TextClip` with `text=None` generates a `TypeError`
```python
TextClip(None, fontsize=30, color ="black")
TypeError: can only concatenate str (not "NoneType") to str
```
So, there should be an error or exception showing that it was not a proper input for the `text` argument.

2. Since `""` (an empty string) is different from `None` and is a `str` type object, there should be a `TextClip` that was generated with empty string.

#### Actual Behavior
When a `TextClip` has parameter of `text=""`, it generates an `OSError`

#### Steps to Reproduce the Problem

You can reproduce the problem using a simple code like below:

```python
from moviepy.editor import *
txt_clip = TextClip("", fontsize=30, color ="black") 
```
Result:
```
OSError: MoviePy Error: creation of None failed because of the following error:

convert: no images defined `PNG32:/tmp/tmp48641d5_.png' @ error/convert.c/ConvertImageCommand/3229.
.

.This error can be due to the fact that ImageMagick is not installed on your computer, or (for Windows users) that you didn't specify the path to the ImageMagick binary in file conf.py, or that the path you specified is incorrect
```

I got every dependencies properly installed on my server as well as had generated `TextClip`s with non empty string. However since the error was confusing and for an end user that is using moviepy for the first time (yes, that is me) kind of wasted time on realizing what was going on for a long time, so I am reporting this issue.

I already kind of fixed this issue by modifying a bit of code, so if Expected Behavior gets chosen, I will pull request.

#### Specifications

  - Python Version: 3.10.4
  - Moviepy Version: 1.0.3
  - Platform Name: Ubuntu Linux
  - Platform Version: 22.04

**Comment by keikoro** [comment]
I didn't have time to take a closer look at your PR, which tries to solve this issue, before, but having read through it now, I'm wondering why we would want to allow TextClip to be used with an empty string in the first place.

I understand that it solves your specific issue, but if TextClip used with an empty string does not result in an error, how will other people be able to catch and debug errors in their scripts that (unintenionally) result in no text showing up (because we silently accept empty strings) when they expect there to be text?

**Comment by keikoro** [comment]
The current error message is obviously not useful, but I think it would make more sense to implement a better error message – or actually catch this error if we are not doing that already (but falling back to some other error message instead) – than silently accepting an empty string... when using TextClip with an empty string is nonsensical/pointless.

Unless there is some use case for that that I'm not aware of?

**Comment by gooday2die** [comment]
## Replys
> I understand that it solves your specific issue, but if TextClip used with an empty string does not result in an error, how will other people be able to catch and debug errors in their scripts that (unintenionally) result in no text showing up (because we silently accept empty strings) when they expect there to be text?

for this, I 100% agree on your opinion. I think I missed this, thanks for pointing this out for me.

## Background story
I was trying to generate a video with caption by`TextClip`. One of those captions had empty string, which was just ``. However, moviepy was throwing `OSError` which I mentioned up in the original issue. Also, since the error was not explicitly saying that no empty string was allowed nor it automatically generated an empty string `TextClip`. Thus, I had to figure out what went wrong and had to replace just empty string into ` ` (strings with a whitespace).

## Solutions that I came up with

Thus, in this issue I mentioned two methods of solving this issue. (At least, I did not wanted anybody else who potentially used an empty string to find out that using emtpy string was not possible and waste time just like me). So I came up with possible behaviors.

1. Letting the user know empty string for `TextClip` is not possible.
2. Silently accept empty string.
3. (Just came up with this idea right now) Generate empty string `TextClip` but warn user that they are generating empty string `TextClip`

## What happened
In the original issue, 
> I already kind of fixed this issue by modifying a bit of code, so if Expected Behavior gets chosen, I will pull request.

was mentioned. However since nobody answered this issue for a long time, I just implemented behavior 2 with the PR that you have seen. 

If you need any further information or need an modification on my PR, feel free to let me know. 
I am more than happy to listen from you. 

Thanks.

**Comment by mondeja** [comment]
> When a TextClip has parameter of text="", it generates an OSError

I can't reproduce it with latest **master**. The problem of passing `None` for `text` and `filename` is being addressed in #1842
